src/Testcontainers.CockroachDb/CockroachDbBuilder.cs

```diff
@@ -0,0 +1,141 @@
+
+    public const string DefaultPassword = "cockroachdb";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CockroachDbBuilder" /> class.
+    /// </summary>
+    public CockroachDbBuilder()
+        : this(new CockroachDbConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CockroachDbBuilder" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    private CockroachDbBuilder(CockroachDbConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+        DockerResourceConfiguration = resourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override CockroachDbConfiguration DockerResourceConfiguration { get; }
+
+    /// <summary>
+    /// Sets the CockroachDb database.
+    /// </summary>
+    /// <param name="database">The CockroachDb database.</param>
+    /// <returns>A configured instance of <see cref="CockroachDbBuilder" />.</returns>
+    public CockroachDbBuilder WithDatabase(string database)
+    {
+        return Merge(DockerResourceConfiguration, new CockroachDbConfiguration(database: database))
+            .WithEnvironment("COCKROACH_DATABASE", database);
+    }
+
+    /// <summary>
+    /// Sets the CockroachDb username.
+    /// </summary>
+    /// <param name="username">The CockroachDb username.</param>
+    /// <returns>A configured instance of <see cref="CockroachDbBuilder" />.</returns>
+    public CockroachDbBuilder WithUsername(string username)
+    {
+        return Merge(DockerResourceConfiguration, new CockroachDbConfiguration(username: username))
+            .WithEnvironment("COCKROACH_USER", username);
+    }
+
+    /// <summary>
+    /// Sets the CockroachDb password.
+    /// </summary>
+    /// <param name="password">The CockroachDb password.</param>
+    /// <returns>A configured instance of <see cref="CockroachDbBuilder" />.</returns>
+    public CockroachDbBuilder WithPassword(string password)
+    {
+        return Merge(DockerResourceConfiguration, new CockroachDbConfiguration(password: password))
+            .WithEnvironment("COCKROACH_PASSWORD", password);
+    }
+
+    /// <inheritdoc />
+    public override CockroachDbContainer Build()
+    {
+        Validate();
+        return new CockroachDbContainer(DockerResourceConfiguration, TestcontainersSettings.Logger);
+    }
+
+    /// <inheritdoc />
+    protected override CockroachDbBuilder Init()
+    {
+        return base.Init()
+            .WithImage(CockroachDbImage)
+            .WithPortBinding(CockroachDbPort, true)
+            .WithPortBinding(CockroachDbRestPort, true)
+            .WithDatabase(DefaultDatabase)
+            .WithUsername(DefaultUsername)
+            .WithPassword(DefaultPassword)
+            .WithCommand("start-single-node")
+            .WithCommand("--insecure")
```

Comment:
`--insecure` is not needed if credentials are set

```suggestion
```

Reply:
I'm thinking to adopt a similar approach as demonstrated in the [CRDB Java](https://github.com/testcontainers/testcontainers-java/blob/main/modules/cockroachdb/src/main/java/org/testcontainers/containers/CockroachContainer.java#L50) repository and use an empty password string while keeping the `---insecure` connection.
What is your opinion on this?

Reply:
sorry for the late reply. Sure, that make sense!

Reply:
Done!
@eddumelendez